packages/internal/src/generate/templates/all-currentUser.d.ts.template

```diff
@@ -3,8 +3,7 @@ import '@redwoodjs/auth'
 
 import { getCurrentUser } from '../../../api/src/lib/auth'
 
-type ThenArg<T> = T extends PromiseLike<infer U> ? U : T
-export type InferredCurrentUser = ThenArg<ReturnType<typeof getCurrentUser>>
+export type InferredCurrentUser = Awaited<ReturnType<typeof getCurrentUser>>
```

Comment:
This has nothing whatsoever to do with the fix. I just stumbled upon this when looking into the CurrentUser declaration merging and couldn't resist simplifying it a bit like this 🙂